src/main/java/org/icgc/argo/program_service/services/EgoService.java

```diff
@@ -43,8 +62,29 @@ public EgoService(@Value("${app.egoPublicKeyUrl}") UrlResource publicKeyResource
     this.egoPublicKey = egoPublicKey;
   }
 
-  public EgoService(RSAPublicKey egoPublicKey) {
+  @Autowired
+  private void setAppJwt(AppProperties appProperties)  {
```

Comment:
what happens when the JWT expires?

Reply:
Good catch, I was about to switch to basic authentication. It's better to be fixed before merge.